packages/hoprd/src/api/v2/paths/channels/{peerid}/{direction}/index.ts

```diff
@@ -1,45 +1,92 @@
 import type { default as Hopr } from '@hoprnet/hopr-core'
 import type { Operation } from 'express-openapi'
 import PeerId from 'peer-id'
-import { STATUS_CODES } from '../../../utils.js'
-import { ChannelInfo, channelStatusToString, formatIncomingChannel, formatOutgoingChannel } from '../index.js'
+import { STATUS_CODES } from '../../../../utils.js'
+import { ChannelInfo, formatIncomingChannel, formatOutgoingChannel } from '../../index.js'
+import {
+  channelStatusToString,
+  ChannelStatus,
+  defer,
+  generateChannelId,
+  PublicKey,
+  type DeferType
+} from '@hoprnet/hopr-utils'
+
+const closingRequests = new Map<string, DeferType<void>>()
 
 /**
  * Closes a channel with provided peerId.
  * @returns Channel status and receipt.
  */
-export const closeChannel = async (node: Hopr, peerIdStr: string, direction: ChannelInfo['type']) => {
+export async function closeChannel(
+  node: Hopr,
+  peerIdStr: string,
+  direction: ChannelInfo['type']
+): Promise<
+  | {
+      success: false
+      reason: keyof typeof STATUS_CODES
+    }
+  | {
+      success: true
+      channelStatus: ChannelStatus
+      receipt: string
+    }
+> {
   let peerId: PeerId
   try {
     peerId = PeerId.createFromB58String(peerIdStr)
   } catch (err) {
     throw Error(STATUS_CODES.INVALID_PEERID)
   }
 
-  const { status: channelStatus, receipt } = await node.closeChannel(peerId, direction)
+  const channelId = generateChannelId(node.getEthereumAddress(), PublicKey.fromPeerId(peerId).toAddress())
+
+  let closingRequest = closingRequests.get(channelId.toHex())
```

Comment:
change here should be part of a different PR

Reply:
Don't think so. Concurrent requests lead to issues. Fixing these issues is the main motivation for this PR